### ISSUE TITLE
Check accessibility of Brand Colors chosen by regular users

### ISSUE THREAD

**Issue by msarit** [forem/forem#17574]
**Describe the bug**
As described in this issue, sometimes users may choose brand colors with poor readability, from an a11y standpoint. For example, choosing "black" (#000000) for `brand_color1` and "black" (#000000) for `brand_color2`.

**To Reproduce**
1. As a user, navigate to your Settings. Scroll down to the Branding section.
2. Choose the same color for both entries and save.
3. The changes are accepted, even though your selections would make some graphics unreadable.

**Expected behavior**
When I choose 2 brand colors that have poor readability, I get a warning and the changes are not persisted.

**Additional Information**
During Onboarding, and within the Admin/Configuration space, we have configured warnings for low accessibility color choices; we could use these implementations as a guide.

**Comment by msarit** [comment]
UPDATE: I am pausing implementation of this ticket to do some research first. In light of this PR, which stopped rendering Follow Buttons using brand colors, I'm not sure that there are still places where we render buttons or other text-on-background using brand colors. In short, this ticket may turn out to be unnecessary.

I will update this ticket after my research.

**Comment by amywtlin** [comment]
@msarit Heads-up that we are removing a redundant brand color: #17686 The expected behavior is still on the right direction, but we will leave only one field for the brand color.

**Comment by msarit** [comment]
@amywtlin Thanks for this update.

Question: Seeing as we're no longer rendering one brand color over another, what now is the purpose for the contrast check?

In my perusal of the codebase, we seem to use User-selected brand color for the background of their profile page (no text displayed) or the borders of their profile Embed (again, no text on the color).

**Comment by amywtlin** [comment]
@msarit thanks for pointing it out. I finally had the capacity to catch up with all details in #17564 and I need to confirm the following with Pawel before I can unblock you, hence I am assigning this issue to Pawel, and will assign back to you after we confirm how we want to proceed.

@ludwiczakpawel: did we decide to not use brand color in the new style follow button for liquid tags anymore?

I am asking because...

1. We [used to use it](https://github.com/forem/forem/issues/17389) (which is while this issue stems from), and I didn't seem to see discussions on stopping using it in #17564 
1. I tried embedding a `user` embed one more time and it is now accessible in the new style, _but_ with the default color, while the user has their own brand color set up (I checked their [user profile](https://dev.to/mhevery))
![Screen Shot 2022-05-18 at 10 16 49 AM](https://user-images.githubusercontent.com/11730007/168943671-1ce56d83-afd4-4239-9dfd-f3d1455a81b6.png)
1. Arit also mentioned in the comment we are only using the brand color for the background of their profile page (no text displayed) or the borders of their profile Embed (again, no text on the color).

Was there a decision made to not use the brand color in the follow button for embeds in the new style? I am fine with either, but I wanted to make sure the interface in production reflects your design decision.

**Comment by ludwiczakpawel** [comment]
> We https://github.com/forem/forem/issues/17389 (which is while this issue stems from), and I didn't seem to see discussions on stopping using it in https://github.com/forem/forem/issues/17564

#17564 is in fact about stopping using it, isn't it?
As mentioned in the description also, the reason why we kept "branded" buttons of liquid tags is because it was a bit hard to update them at the time. But the intention was to deprecate branded buttons and only use the ones from Crayons. 

> Was there a decision made to not use the brand color in the follow button for embeds in the new style? I am fine with either, but I wanted to make sure the interface in production reflects your design decision.

As mentioned above: there was a decision to only use Crayons buttons whenever possible and stop using ones with profile brand colors. I don't think this decision has been documented anywhere in the past except the few issues/PRs that could be considered as evidence. 

Re-assigning back to you @amywtlin .
